firefox-ios/Client/Frontend/Browser/LegacyClipboardBarDisplayHandlerDelegate.swift

```diff
@@ -0,0 +1,107 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+import Common
+
+// TODO: FXIOS-TODO This legacy ClipboardBar code can be removed once iOS 15 is dropped
+protocol LegacyClipboardBarDisplayHandlerDelegate: AnyObject {
+    func shouldDisplay(clipBoardURL url: URL)
+}
+
+@preconcurrency
+final class LegacyClipboardBarDisplayHandler: ClipboardBarDisplayHandler {
+    struct UX {
+        static let toastDelay = DispatchTimeInterval.milliseconds(10000)
+    }
+
+    weak var delegate: LegacyClipboardBarDisplayHandlerDelegate?
+    weak var tabManager: TabManager?
```

Comment:
That `tabManager` was making it harder to have the class `@MainActor`, hence why splitting this code into a legacy class was also making my life easier 😄 `tabManager` is only used in the iOS 15 code path.